test-cloud/test-cloud-performance/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iJbpmEJBTimerPerfIntegrationTest.java

```diff
@@ -129,69 +133,59 @@
+        limits.put(CPU, System.getProperty("limits.cpu","4000m"));
+        limits.put(MEMORY, System.getProperty("limits.memory","4Gi"));
+        deploymentScenario.getKieServerDeployment().setResources(requests, limits);
+
+        if (SCALE_COUNT >= 1) {
             logger.info("starting to scale");
             scaleKieServerTo(SCALE_COUNT);
             logger.info("scaled to {} pods", SCALE_COUNT);
-        } else if (SCALE_COUNT == 1) {
-            pods.add(deploymentScenario.getKieServerDeployments().get(0).getInstances().get(0).getName());
         } else {
             throw new RuntimeException("wrong scale parameter, should be equal or greater than 1");
         }
         
+        deploymentScenario.getKieServerDeployment().setRouterTimeout(Duration.ofMinutes(ROUTER_TIMEOUT));
+        deploymentScenario.getKieServerDeployment().setRouterBalance(ROUTER_BALANCE);
         
-        deploymentScenario.getKieServerDeployments().get(0).setRouterTimeout(Duration.ofMinutes(ROUTER_TIMEOUT));
-        deploymentScenario.getKieServerDeployments().get(0).setRouterBalance(ROUTER_BALANCE);
-        
-        Map<String, String> requests = new HashMap<String, String>();
-        requests.put(CPU, System.getProperty("requests.cpu","1000m"));
-        requests.put(MEMORY, System.getProperty("requests.memory","1Gi"));
-        Map<String, String> limits = new HashMap<String, String>();
-        limits.put(CPU, System.getProperty("limits.cpu","4000m"));
-        limits.put(MEMORY, System.getProperty("limits.memory","4Gi"));
-        deploymentScenario.getKieServerDeployments().get(0).setResources(requests, limits);
-        
-        //changing DC implies pods are also restarted, so to be sure we need to wait for this to be ready
-        deploymentScenario.getKieServerDeployments().get(0).waitForScale();
-        kieServicesClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployments().get(0), Duration.ofMinutes(60).toMillis());
+        kieServicesClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment(), Duration.ofMinutes(60).toMillis());
         
         logger.info("Setting timeout for kieServerClient to {}", Duration.ofMinutes(60).toMillis());
         
-        processServicesClient = KieServerClientProvider.getProcessClient(deploymentScenario.getKieServerDeployments().get(0));
-        queryServicesClient = KieServerClientProvider.getQueryClient(deploymentScenario.getKieServerDeployments().get(0));
+        processServicesClient = KieServerClientProvider.getProcessClient(deploymentScenario.getKieServerDeployment());
+        queryServicesClient = KieServerClientProvider.getQueryClient(deploymentScenario.getKieServerDeployment());
     }
 
     @After
-    public void deleteRepo() {
-        Git.getProvider().deleteGitRepository(repositoryName);
-        deploymentScenario.getKieServerDeployments().get(0).resetRouterTimeout();        
```

Comment:
Is it not longer needed to reset the router timeout for the deployment?

Reply:
Timer reset is done in resetRouterTimeout() below :)

Reply:
sorry, haven't seen it :)